src/index.ts

```diff
@@ -223,12 +77,50 @@ export class CalimeroSdk {
     clearLocalStorage();
     window.location.href = "/";
   }
+
+  setCredentials = () =>  {
+    if(window.location.hash) {
+      try{
+        const decodedData = JSON.parse(decodeURIComponent(window.location.hash.substring(1)));
+        const walletData = JSON.parse(decodedData.calimeroToken).walletData;
+        const message = walletData.message;
+        const accountId = walletData.accountId;
+        const publicKey = walletData.publicKey.data.data;
+        const authToken = decodedData.secretToken;
+        const sentHash = localStorage.getItem(MESSAGE_HASH_KEY);
+        if(message !== sentHash){
+          throw new Error("Sent Message hash is not equal to receiver, please try again!");
+        }
+
+        localStorage.setItem(AUTH_TOKEN_KEY,
+          authToken);
+        localStorage.setItem(MESSAGE_KEY,
+          message);
+        localStorage.setItem(ACCOUNT_ID,
+          accountId);
+        localStorage.setItem(PUBLIC_KEY,
+          JSON.stringify(publicKey));
+        this.confirmSignIn();
+        return {success: "Sign in confirmed!"};
+
+      }catch(error){
+        if (typeof error === "string") {
+          return {error: error.toUpperCase()}
+        } else if (error instanceof Error) {
+          return {error: error.message}
+        }
+      } 
+    }
+  }
+
+  confirmSignIn = () => {
+    if(window.location.hash){
+      window.location.replace(window.location.origin)
```

Comment:
what about other query params?

Reply:
window.location.replace(window.location.origin+window.location.pathname);
Should cover it 
 source: https://ui.dev/get-current-url-javascript